thetis/solver2d.py

```diff
@@ -577,7 +577,7 @@ def iterate(self, update_forcings=None,
 
             # Move to next time step
             self.iteration += 1
-            self.simulation_time = self.iteration*self.dt
+            self.simulation_time += self.dt
```

Comment:
Hmm, the problem here is that if you accumulate `dt` to `simulation_time` floating point errors will accumulate. This can be an issue if (read: when) you do long runs with small time steps. File exports etc. are done by comparing the simulation time floats, so you may end up exporting one time step too early/late, which can screw up analysis in postprocessing (e.g. tides). Using the multiplication avoids the drift in time precision. You could fix this by adding another float `initial_time` that defaults to zero.

Reply:
Ok, I have done that! I note however that `time += dt` is the way the time is updated in 3D, so is it a real problem ?

Reply:
Thanks, looks good. Good point about the 3D solver, I think we should change that too.